flow/layers/image_filter_layer.cc

```diff
@@ -117,6 +117,13 @@ void ImageFilterLayer::Paint(PaintContext& context) const {
     }
   }
 
+  // Only apply the offset if not being raster-cached to avoid the offset being
+  // applied twice.
+  mutator.translate(offset_);
+  if (context.raster_cache) {
+    mutator.integralTransform();
+  }
```

Comment:
I don't like this reorganization of the integral Transform. It's similar, but not exactly the math that should be happening.

Reply:
I did this because the raster cache bounds get rounded-out. So if the child is at 99.9, IntegralTransform will snap it to 100, but the raster cache will capture from 99. It doesn't cause a position mismatch, but it means you get 1 blank row and/or column raster cached.

Reply:
Hm where should the rounding go? I put it here to preserve the fractional part and only round at the end. But I guess that isn't what happens in Preroll, it's rounded before going through the image filter.